libbeat/common/version.go

```diff
@@ -150,3 +151,23 @@ func (v *Version) metaIsLessThanOrEqual(v1 *Version) bool {
 	}
 	return v.Meta <= v1.Meta
 }
+
+// UnmarshalJSON unmarshals a JSON string version representation into a Version struct
+// Implements https://golang.org/pkg/encoding/json/#Unmarshaler
+func (v *Version) UnmarshalJSON(version []byte) error {
+	var versionStr string
+	err := json.Unmarshal(version, &versionStr)
+	if err != nil {
+		return err
+	}
+
+	ver, err := NewVersion(versionStr)
+
+	v.version = versionStr
+	v.Major = ver.Major
+	v.Minor = ver.Minor
+	v.Bugfix = ver.Bugfix
+	v.Meta = ver.Meta
+
+	return err
```

Comment:
If NewVersion fails can return a nil `ver`? then these assignations would fail, I think we should check the error before assigning.

Also once error is checked, I think we could assign the values by `*v = *ver`.